apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -320,6 +323,34 @@ export default class ProjectManager {
     this.lastSource = JSON.stringify(lastSource);
   }
 
+  getShouldCaptureThumbnail() {
+    const {isShareView} = getStore().getState().lab;
+    return this.channelId && this.lastChannel?.isOwner && !isShareView;
+  }
+
+  setThumbnail(pngBlob: Blob) {
+    this.thumbnailPngBlob = pngBlob;
+    this.thumbnailUrl = `/v3/files/${this.channelId}/.metadata/thumbnail.png`;
+  }
+
+  /**
+   * Uploads a thumbnail image to the thumbnail path via the files API.
+   */
+  saveThumbnail(): Promise<Response | void> {
+    if (this.thumbnailUrl && this.thumbnailPngBlob) {
+      return HttpClient.put(
```

Comment:
This can throw an error right? We should catch the error so it doesn't crash anything